src/Core/Grid/Query/ManufacturerAddressQueryBuilder.php

```diff
@@ -0,0 +1,152 @@
+ */
+final class ManufacturerAddressQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @var DoctrineSearchCriteriaApplicatorInterface
+     */
+    private $searchCriteriaApplicator;
+
+    /**
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
+     * @param int $contextLangId
+     */
+    public function __construct(
+        Connection $connection,
+        $dbPrefix,
+        DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator,
+        $contextLangId
+    ) {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->searchCriteriaApplicator = $searchCriteriaApplicator;
+        $this->contextLangId = $contextLangId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilderByFilters($searchCriteria->getFilters());
+        $qb->select('a.id_address, m.name, a.firstname, a.lastname, a.postcode, a.city, cl.name as country');
+
+        $this->searchCriteriaApplicator
+            ->applyPagination($searchCriteria, $qb)
+        ;
+
+        return $qb;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilderByFilters($searchCriteria->getFilters());
+        $qb->select('COUNT(DISTINCT a.`id_address`)');
+
+        return $qb;
+    }
+
+    /**
+     * Gets query builder with common sql needed for manufacturer addresses grid.
+     *
+     * @param array $filters
+     *
+     * @return QueryBuilder
+     */
+    private function getQueryBuilderByFilters(array $filters)
+    {
+        $allowedFilters = [
+            'id_address',
+            'brand',
+            'firstname',
+            'lastname',
+            'postcode',
+            'city',
+            'country',
+        ];
+        $qb = $this->connection
+            ->createQueryBuilder()
+            ->from($this->dbPrefix . 'address', 'a')
+            ->leftJoin(
+                'a',
+                $this->dbPrefix . 'country_lang',
+                'cl',
+                'cl.id_country = a.id_country AND cl.id_lang = :lang'
+            )
+            ->setParameter('lang', $this->contextLangId)
+            ->leftJoin(
+                'a',
+                $this->dbPrefix . 'manufacturer',
+                'm', 'm.id_manufacturer = a.id_manufacturer'
+            )
+            ->andWhere('a.id_customer = 0')
+            ->andWhere('a.id_supplier = 0')
+            ->andWhere('a.id_warehouse = 0')
+            ->andWhere('a.deleted = 0')
+        ;
+        //@todo: Enable filtering when its fixed in another PR
```

Comment:
Is it related to double-grid filtering ?

Reply:
Yes